pimd/pim_igmp.c

```diff
@@ -467,6 +467,8 @@ int pim_igmp_packet(struct igmp_sock *igmp, char *buf, size_t len)
 		int max_resp_code = igmp_msg[1];
 		int query_version;
 
+		if (igmp->mtrace_only)
```

Comment:
so the reason I asked for this test to be put inside the functions is that they would just return with an appropriate error return from this function.

Reply:
OK. I didn't get your comment the first time. Got it now.